src/rgw/rgw_threadpool.h

```diff
@@ -0,0 +1,121 @@
+#ifndef _RGW_THREADPOOL_
+#define _RGW_THREADPOOL_
+
+#include <queue>
+
+class WorkerThread {
```

Comment:
hmm; doesn't RGW already have threadpools?

Reply:
why do we need a thread pool? aren't we using async io (via aio and libcurl) everywhere?

Reply:
we are using `libcurl 'curl_easy'` interface. The easy interface is a synchronous and a blocking function.

Reply:
other code in rgw sends http requests with `RGWHTTPClient` and `RGWHTTPManager`, which uses the `curl_multi` interface to do this with a single thread. we'll want to use that instead

and if this is for l2, let's move it to the other PR